File: solutions/scala/accumulate/1/src/main/scala/Accumulate.scala
class Accumulate {
  def accumulate[A, B](
                        function: A => B,
                        list: List[A],
                        accumulated: List[B] = List[B]()
                      ): List[B] = {
    if (list.isEmpty) {
      accumulated.reverse
    }
    else {
      this.accumulate(function, list.tail, function(list.head) :: accumulated)
    }
  }
}

File: solutions/erlang/accumulate/1/src/accumulate.erl
-module(accumulate).

-export([accumulate/2]).

accumulate(Function, List) ->
  accumulate(Function, List, _Accumulated = []).

accumulate(_Function, List, Accumulated)
  when length(List) == 0 ->
  lists:reverse(Accumulated);

accumulate(Function, [Head | Tail], Accumulated) ->
  accumulate(Function, Tail, [Function(Head) | Accumulated]).